cmd/kindnetd/subnets.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+	"regexp"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type subnets struct {
+	pods sets.String
+	services sets.String
+}
+
+func newSubnets() *subnets {
+	return &subnets{
+		pods: sets.NewString(),
+		services: sets.NewString(),
+	}
+}
+
+func (s *subnets) All() []string {
+	return s.pods.Union(s.services).List()
+}
+
+
+func getNoMasqueradeSubnets(clientset *kubernetes.Clientset) []string {
+	s := newSubnets()
+
+	// check for environment variables (legacy)
```

Comment:
maybe we should start considering dual stack, this function it will accept 1 cidr or multiple in a comma separated list https://github.com/khenidak/kubernetes/blob/0fbfa755d3960eaa0ab3ea38453a934d2188e7f8/cmd/cloud-controller-manager/app/core.go#L144